### PR TITLE
Simple GUI test for CI

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
Closes https://github.com/computational-cell-analytics/micro-sam/issues/205

This PR:
1. Updates the github test.yaml workflow file to enable GUI testing on CI (as [described here](https://github.com/computational-cell-analytics/micro-sam/issues/205#issuecomment-1736683856)), and 
2. Adds a very simple GUI test which opens the 2D annotator dock widgets in napari. We don't interact with it or do any actual testing of the annotation functions. This GUI test just creates the image embedding and then opens the dock widgets in the napari viewer, as if it was ready to start.

That brings our [code coverage up to 23%](https://app.codecov.io/github/GenevieveBuckley/micro-sam/commit/58aeaa7978da616d6240ed42b222680f0361145c/tree). I did do some extra work (not in this branch) to see if I could mimic user interaction of annotation to this test... but that only increased the test coverage by about 2%. My guess is that these functions might already be covered by unit tests, so it's not helpful to make a giant integration test anyway.

**Careful**: this PR is just added straight on top of the many, many git commits involved in branch https://github.com/computational-cell-analytics/micro-sam/pull/214. We should do, uh, something with git to fix any problems from that. (Worst case scenario, I can close and and reopen a new PR for this after https://github.com/computational-cell-analytics/micro-sam/pull/214 is merged.)

### CODE REVIEW

**Comment by GenevieveBuckley** [comment]
I really do not like how the CI job takes <5 minutes on the `codecov-report` PR branch, but on this branch it is taking over half an hour.
Trying to figure out why it's so slow:
* I don't think it's installing pyqt and napari, all the steps before we run pytest finish in a reasonable amount of time
* The `pytest` step is very, very slow!
    * It's not because my new GUI test is slow - I commented that out in one of the commits, but still found the CI job took an unreasonably long time to run
    * ...I think it might be the use of `aganders3/headless-gui`? We could test this theory by replacing `pytest` with `echo hello` and then look at how long each CI step took to run. I'm not sure what to do if this is the reason the CI is slow, I think we need it for GUI tests.

**Comment by GenevieveBuckley** [comment]
Update: I think it's the `test/test_training.py` file that was making the CI so slow. I accidentally opened this PR against the master branch, not dev (not that it helps much, because everything needs to work in master eventually)

A possible workaround would be to mark the super slow tests as `pytest.mark.slow` and skip them in the CI jobs.

**Comment by constantinpape** [comment]
> Update: I think it's the `test/test_training.py` file that was making the CI so slow.

Yes, unfortunately `test_training.py` is quite slow.

> A possible workaround would be to mark the super slow tests as `pytest.mark.slow` and skip them in the CI jobs.

Yes, that's one possibility. Before that I want to check if I can improve this test a bit to speed it up on CI and so that we can just run it with the normal tests.
(I developed the test on my notebook, and it was fast enough, but I guess the CPU in the CI is weaker, so that it takes quite long.)

For now: since you now have the PR on dev it should not be a problem here. Can you go ahead and fix the merge conflicts (which I think are because I squashed and merged #214) and, if you have any more changes you want to make, add them too?

I will then:
- squash and merge this PR
- merge `dev` into `master` and also update `dev`, so that we have all the tests up-to-date and on the same branches
- look into speeding up `test_training.py` (or mark it as `pytest.mark.slow` if I can't speed it up without making the test worse)
- look into adding more GUI tests

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
   test:
     name: ${{ matrix.os }} ${{ matrix.python-version }}
     runs-on: ${{ matrix.os }}
-
+    timeout-minutes: 60
```
Comment (GenevieveBuckley) [inline .github/workflows/test.yaml:18]: Right now, the CI job finishes in around six-ish minutes. On the master branch, including the `test_training.py` script, it takes over half an hour. Google says the default github actions timeout is 360 minutes (6 hours)... but we were seeing CI jobs killed with an error much sooner than that on the master branch (maybe they ran out of memory?)

I'm not sure whether we should keep this line, remove it, or change it to a different number. I'm leaning towards keeping it in here, and maybe changing the timeout number if that makes sense.

**Comment by GenevieveBuckley** [comment]
I've fixed the merge conflicts, and opened a separate issue for discussion about `test_training.py`.

I'm very happy that this CI job runs in about 6 minutes - the gui test is not slow at all!

File: .github/workflows/test.yaml (hunk 1 above)
Comment (constantinpape) [reply .github/workflows/test.yaml:18]: Let's keep it for now. I will look into the test.

**Pull request merged**